sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -340,7 +341,7 @@ static size_t _handle_req(coap_pkt_t *pdu, uint8_t *buf, size_t len,
     gcoap_observe_memo_t *memo          = NULL;
     gcoap_observe_memo_t *resource_memo = NULL;
 
-    switch (_find_resource(pdu, &resource, &listener)) {
+    switch (_find_resource((const coap_pkt_t *)pdu, &resource, &listener)) {
```

Comment:
The constness could also be propagated up.

Reply:
Not sure about this.  Propagating the constness might require to change the signature of many other functions.  Let's do the assessment and a potential change in another PR.